nemo/core/connectors/save_restore_connector.py

```diff
@@ -479,37 +481,35 @@ def _handle_artifacts(self, model, nemo_file_folder):
             # Need to step into nemo archive to extract file
             # Get path where the command is executed - the artifacts will be "retrieved" there
             # (original .nemo behavior)
-            cwd = os.getcwd()
-            # Step into the nemo archive to try and find the file
-            # TemporaryDirectory context must always be outer to try-catch chdir otherwise it crashes on Windows
-            with tempfile.TemporaryDirectory() as archive_dir:
-                try:
-                    # unpack all restorations paths (nemo checkpoints)
-                    # in nemo checkpoints all resources contain hash in name, so there should be no collisions
-                    for path in restoration_paths:
-                        if self.model_extracted_dir:
-                            shutil.copytree(src=path, dst=archive_dir, dirs_exist_ok=True)
-                        else:
-                            self._unpack_nemo_file(path2file=path, out_folder=archive_dir)
-                    os.chdir(archive_dir)
-                    for conf_path, artiitem in tarfile_artifacts:
-                        # Get basename and copy it to nemo_file_folder
-                        if 'nemo:' in artiitem.path:
-                            artifact_base_name = artiitem.path.split('nemo:')[1]
-                        else:
-                            artifact_base_name = os.path.basename(artiitem.path)
-                        # no need to hash here as we are in tarfile_artifacts which are already hashed
-                        artifact_uniq_name = artifact_base_name
-                        shutil.copy2(artifact_base_name, os.path.join(nemo_file_folder, artifact_uniq_name))
-
-                        # Update artifacts registry
-                        new_artiitem = model_utils.ArtifactItem()
-                        new_artiitem.path = "nemo:" + artifact_uniq_name
-                        new_artiitem.path_type = model_utils.ArtifactPathType.TAR_PATH
-                        model.artifacts[conf_path] = new_artiitem
-                finally:
-                    # change back working directory
-                    os.chdir(cwd)
+            for conf_path, artiitem in tarfile_artifacts:
```

Comment:
These changes aren't required for the scope of just supporting different file from being extracted.

Reply:
Could you elaborate on why this change isn't needed in this "scope"? Do you mean I should move this logic elsewhere? If so, could you advise.

Or are you saying this change isn't required at all? I tried outlining the issue I observed in the description. This change is to avoid naively copying huge checkpoints just to extract a few small artifact files.